codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/HuffmanDecoder.java

```diff
@@ -213,4 +148,109 @@ private static void insert(Node root, int symbol, int code, byte length) {
             current.children[i] = terminal;
         }
     }
+
+    private final class Decoder {
+        private final DecoderProcessor processor = new DecoderProcessor();
+        public byte[] decode(ByteBuf buffer, int length) throws IOException {
+            try {
+                buffer.forEachByte(buffer.readerIndex(), length, processor);
+                buffer.skipBytes(length);
+                return processor.end();
+            } finally {
+                processor.reset();
+            }
+        }
+
+        private final class DecoderProcessor implements ByteProcessor {
+            private final ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
The only thing I not like about ByteArrayOutputStream is all the `synchronized(...)` going on there. Maybe we should just use byte[] and handle the expanding etc by ourself. @nmittler @Scottmitch WDYT ?